smartsim/entity/model.py

```diff
@@ -242,500 +253,32 @@ def print_attached_files(self) -> None:
-        function: t.Optional[str] = None,
-        device: str = Device.CPU.value.upper(),
-        devices_per_node: int = 1,
-        first_device: int = 0,
-    ) -> None:
-        """TorchScript function to launch with this Application instance
-
-        Each script function to the application will be loaded into a
-        non-converged FeatureStore prior to the execution
-        of this Application instance.
-
-        For converged featurestores, the :meth:`add_script` method should be used.
-
-        Device selection is either "GPU" or "CPU". If many devices are
-        present, a number can be passed for specification e.g. "GPU:1".
-
-        Setting ``devices_per_node=N``, with N greater than one will result
-        in the application being stored in the first N devices of type ``device``.
-
-        :param name: key to store function under
-        :param function: TorchScript function code
-        :param device: device for script execution
-        :param devices_per_node: The number of GPU devices available on the host.
-               This parameter only applies to GPU devices and will be ignored if device
-               is specified as CPU.
-        :param first_device: The first GPU device to use on the host.
-               This parameter only applies to GPU devices and will be ignored if device
-               is specified as CPU.
-        """
-        fs_script = FSScript(
-            name=name,
-            script=function,
-            device=device,
-            devices_per_node=devices_per_node,
-            first_device=first_device,
-        )
-        self.add_script_object(fs_script)
-
-    def __hash__(self) -> int:
-        return hash(self.name)
-
-    def __eq__(self, other: object) -> bool:
-        if not isinstance(other, Application):
-            return False
-
-        if self.name == other.name:
-            return True
-        return False
+        ...
 
     def __str__(self) -> str:  # pragma: no cover
         entity_str = "Name: " + self.name + "\n"
         entity_str += "Type: " + self.type + "\n"
-        entity_str += str(self.run_settings) + "\n"
-        if self._fs_models:
-            entity_str += "FS Models: \n" + str(len(self._fs_models)) + "\n"
-        if self._fs_scripts:
-            entity_str += "FS Scripts: \n" + str(len(self._fs_scripts)) + "\n"
-        return entity_str
+        entity_str += "Executable:\n"
+        for ex in self.exe:
+            entity_str += f"{ex}\n"
+        entity_str += "Executable Arguments:\n"
+        for ex_arg in self.exe_args:
+            entity_str += f"{str(ex_arg)}\n"
+        entity_str += "Entity Files: " + str(self.files) + "\n"
+        entity_str += "File Parameters: " + str(self.file_parameters) + "\n"
+        entity_str += "Incoming Entities:\n"
+        for entity in self.incoming_entities:
+            entity_str += f"{entity}\n"
+        entity_str += "Key Prefixing Enabled: " + str(self.key_prefixing_enabled) + "\n"
```

Comment:
Nit: Can we make this f-strs?
```suggestion
        entity_str += f"Entity Files: {self.files}\n"
        entity_str += f"File Parameters: {self.file_parameters}\n"
        entity_str += "Incoming Entities:\n"
        for entity in self.incoming_entities:
            entity_str += f"{entity}\n"
        entity_str += f"Key Prefixing Enabled: {self.key_prefixing_enabled}\n"
```

Reply:
It also might be worth making this a `io.StringIO` or a big `textwrap.dedent` to avoid the manual string concatenation, but its an incredibly minor micro-optimization

Reply:
changed to f strings